lib/cloud/azure/vm.go

```diff
@@ -166,3 +167,74 @@ func (c *vmClient) ListVirtualMachines(ctx context.Context, resourceGroup string
 
 	return virtualMachines, nil
 }
+
+// vmRunCommand provides an interface for an Azure Run Command client.
+type vmRunCommand interface {
+	BeginCreateOrUpdate(ctx context.Context, resourceGroup, vmName, runCommandName string, runCommand armcompute.VirtualMachineRunCommand, options *armcompute.VirtualMachineRunCommandsClientBeginCreateOrUpdateOptions) (*runtime.Poller[armcompute.VirtualMachineRunCommandsClientCreateOrUpdateResponse], error)
```

Comment:
There are too many parameters in this method, and too many of them weakly type as well (strings). This tends to make call sites confusing/ambiguous to read.

Please define a struct to hold the parameters (except from `ctx`), so we can use struct initialization to document call sites.